gms/gossiper.cc

```diff
@@ -1388,13 +1388,13 @@ endpoint_state& gossiper::get_endpoint_state(inet_address ep) {
 }
 
 future<> gossiper::reset_endpoint_state_map() {
-    if (this_shard_id() != 0) {
-        on_internal_error(logger, "must be called on shard 0");
-    }
-    _unreachable_endpoints.clear();
-    _live_endpoints.clear();
-    co_await update_live_endpoints_version();
-    co_await container().invoke_on_all([] (gossiper& g) {
+    logger.debug("Resetting endpoint state map");
+    auto lock = lock_endpoint_update_semaphore();
+    auto version = _live_endpoints_version + 1;
+    co_await container().invoke_on_all([version] (gossiper& g) {
```

Comment:
> Clearing them now would get rid of potential redundant replication on the next gossiper round.

More than that. It can actually prevent replication by mistake if _live_endpoints is updated to become equal to _shadow_live_endpoints (as was after the shadow round), so it's an actual bug.

Opened https://github.com/scylladb/scylladb/issues/15003